src/solve.jl

```diff
@@ -176,14 +176,33 @@ function solve!(system::InitialValueProblem{<:HybridSystem,
              inout_map = reach_tube.options[:inout_map]  # TODO temporary hack
         end
 
+
         # get the property for the current location
         property_loc = property isa Dict ?
                        get(property, loc_id, nothing) :
                        property
         if property_loc != nothing
-            for (i, reach_set) in enumerate(reach_tube.Xk)
-                if !check(property_loc, reach_set.X)
-                    return CheckSolution(false, i, options)
+            if opD isa DecomposedDiscretePost
+                temp_vars = opD.options[:temp_vars]
+                n_lowdim = length(temp_vars)
+                property_loc_lowdim = project_property(property_loc, temp_vars)
+                property_loc_lowdim = Properties.project(property_loc, temp_vars)
```

Comment:
```suggestion
                property_loc_lowdim = project(property_loc, temp_vars)
```
Again, this should not be necessary.